src/main/java/core/basesyntax/service/impl/DataConverterImpl.java

```diff
@@ -0,0 +1,35 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.DataConverter;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> data) {
+        List<FruitTransaction> transactions = new ArrayList<>();
+        boolean isHeader = true; // Skip the header line
+        for (String line : data) {
+            if (isHeader) {
+                isHeader = false;
+                continue;
+            }
+            String[] parts = line.split(",");
+            if (parts.length != 3) {
+                throw new IllegalArgumentException("Invalid data format");
+            }
+            try {
```

Comment:
This try-catch block is catching an IllegalArgumentException only to throw another IllegalArgumentException. Consider whether this is necessary, and if so, ensure that the new exception provides additional context or clarification over the original exception.

Reply:
fix this, you don`t need to put all this code into try block